src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3525,19 +3532,15 @@ CoPartitionedTables(List *firstShardList, List *secondShardList)
  * ShardIntervalsEqual checks if given shard intervals have equal min/max values.
  */
 static bool
-ShardIntervalsEqual(ShardInterval *firstInterval, ShardInterval *secondInterval)
+ShardIntervalsEqual(FmgrInfo* comparisonFunction, ShardInterval *firstInterval,
```

Comment:
I think we need to update the function comment to reflect the comparisonFunction parameter